$NetBSD: patch-Makefile,v 1.3 2025/10/07 20:45:10 dkazankov Exp $

Fix target calculation
Add necessary parameters for pkgsrc build
Fix gnatcoll.gpr installation
Fix destdir building in rsync absence
Make test target

--- Makefile.orig	2024-09-24 12:28:32.000000000 +0300
+++ Makefile
@@ -61,7 +61,7 @@
 GNATCOLL_GPR=$(SOURCE_DIR)/gnatcoll.gpr
 
 TARGET := $(shell gcc -dumpmachine)
-NORMALIZED_TARGET := $(subst normalized_target:,,$(wordlist 6,6,$(shell gprconfig  --config=ada --target=$(TARGET) --mi-show-compilers)))
+NORMALIZED_TARGET := $(lastword $(subst :, ,$(shell gprconfig  --config=ada --target=$(TARGET) --mi-show-compilers | grep ' 1 normalized_target:')))
 ifeq ($(NORMALIZED_TARGET),)
   $(error No toolchain found for target "$(TARGET)")
 endif
@@ -75,6 +75,7 @@
 ENABLE_SHARED = yes
 INTEGRATED    = no
 GNATCOV       =
+TESTSUITE_DIR := $(shell pwd)/testsuite
 
 all: build
 
@@ -98,8 +99,13 @@
  --build=$(BUILD) \
  --target=$(NORMALIZED_TARGET) \
  --prefix=local-install \
- --install \
- --enable-shared=$(ENABLE_SHARED)
+ --enable-shared=$(ENABLE_SHARED) \
+ --add-gpr-path='./minimal,./core,./projects' \
+ --gpr-opts $(GPRBUILD_OPTIONS)
+
+INSTALL_ARGS= \
+ --prefix=local-install \
+ --gpr-opts $(GPRINSTALL_OPTIONS)
 
 build:
 	rm -rf local-install
@@ -112,18 +118,28 @@
 endif
 
 	$(PYTHON) $(SOURCE_DIR)/minimal/gnatcoll_minimal.gpr.py build $(INSTR_BUILD_OPTS) $(BUILD_ARGS)
+	$(PYTHON) $(SOURCE_DIR)/minimal/gnatcoll_minimal.gpr.py install $(INSTALL_ARGS)
 
 ifeq ($(GNATCOLL_MINIMAL_ONLY), no)
 	$(PYTHON) $(SOURCE_DIR)/core/gnatcoll_core.gpr.py build $(INSTR_BUILD_OPTS) $(BUILD_ARGS)
+	$(PYTHON) $(SOURCE_DIR)/core/gnatcoll_core.gpr.py install $(INSTALL_ARGS)
 endif
 
 ifeq ($(GNATCOLL_PROJECTS), yes)
 	$(PYTHON) $(SOURCE_DIR)/projects/gnatcoll_projects.gpr.py build $(INSTR_BUILD_OPTS) $(BUILD_ARGS)
+	$(PYTHON) $(SOURCE_DIR)/projects/gnatcoll_projects.gpr.py install $(INSTALL_ARGS)
 endif
 
+	gprinstall -f -P local-install/share/gpr/gnatcoll.gpr -p --prefix=local-install/
+
 install:
 	@echo "Installing gnatcoll into $(prefix)"
-	rsync -av ./local-install/ $(prefix)$(integrated_install)  
+	cp -r ./local-install/* $(prefix)$(integrated_install)  
+
+test:
+	$(PYTHON) testsuite/run-tests --recompile \
+	    -t $(TESTSUITE_DIR)/temp -d $(TESTSUITE_DIR)/dest -o $(TESTSUITE_DIR)/out \
+	    --no-random-temp-subdir --cleanup-mode none --show-error-output
 
 ###########
 # Cleanup #
